storage/grpc_metrics.go

```diff
@@ -0,0 +1,295 @@
+		log.Printf("The Project ID configured for metrics is %s, but the Project ID of the storage client is %s. Make sure that the service account in use has the required metric writing role (roles/monitoring.metricWriter) in the project projectIdToUse or metrics will not be written.", preparedResource.projectToUse, config.project)
+	}
+	meOpts := []mexporter.Option{
+		mexporter.WithProjectID(preparedResource.projectToUse),
+		mexporter.WithMetricDescriptorTypeFormatter(metricFormatter),
+		mexporter.WithCreateServiceTimeSeries(),
+		mexporter.WithMonitoredResourceDescription(monitoredResourceName, []string{"project_id", "location", "cloud_platform", "host_id", "instance_id", "api"})}
+	endpointToUse := determineMonitoringEndpoint(config.endpoint)
+	if endpointToUse != "" {
+		meOpts = append(meOpts, mexporter.WithMonitoringClientOptions(option.WithEndpoint(endpointToUse)))
+	}
+	exporter, err := mexporter.New(meOpts...)
+	if err != nil {
+		return nil, err
+	}
+	metricViews := []metric.View{
+		createHistogramView("grpc.client.attempt.duration", "A view of grpc.client.attempt.duration with histogram boundaries more appropriate for Google Cloud Storage RPCs", "s", latencyHistogramBoundaries()),
+		createHistogramView("grpc.client.attempt.rcvd_total_compressed_message_size", "A view of grpc.client.attempt.rcvd_total_compressed_message_size with histogram boundaries more appropriate for Google Cloud Storage RPCs", "By", sizeHistogramBoundaries()),
+		createHistogramView("grpc.client.attempt.sent_total_compressed_message_size", "A view of grpc.client.attempt.sent_total_compressed_message_size with histogram boundaries more appropriate for Google Cloud Storage RPCs", "By", sizeHistogramBoundaries()),
+	}
+	provider := metric.NewMeterProvider(
+		metric.WithReader(metric.NewPeriodicReader(&exporterLogSuppressor{exporter: exporter}, metric.WithInterval(time.Minute))),
+		metric.WithResource(preparedResource.resource),
+		metric.WithView(metricViews...),
+	)
+	mo := opentelemetry.MetricsOptions{
+		MeterProvider: provider,
+		Metrics: opentelemetry.DefaultMetrics().Add(
+			"grpc.lb.wrr.rr_fallback",
+			"grpc.lb.wrr.endpoint_weight_not_yet_usable",
+			"grpc.lb.wrr.endpoint_weight_stale",
+			"grpc.lb.wrr.endpoint_weights",
+			"grpc.lb.rls.cache_entries",
+			"grpc.lb.rls.cache_size",
+			"grpc.lb.rls.default_target_picks",
+			"grpc.lb.rls.target_picks",
+			"grpc.lb.rls.failed_picks"),
+		OptionalLabels: []string{"grpc.lb.locality"},
+	}
+	do := option.WithGRPCDialOption(opentelemetry.DialOption(opentelemetry.Options{MetricsOptions: mo}))
+	context := &internalMetricsContext{
+		project:    preparedResource.projectToUse,
+		endpoint:   endpointToUse,
+		clientOpts: []option.ClientOption{do},
+		provider:   provider,
+		close:      createShutdown(ctx, provider),
+	}
+	return context, nil
+}
+
+func enableClientMetrics(ctx context.Context, s *settings) (*internalMetricsContext, error) {
+	_, ep, err := htransport.NewClient(ctx, s.clientOption...)
```

Comment:
It seems kind of backwards that we need to initialize a new http client just in order to capture the endpoint. Am I reading this correctly? Is there not another way to get this endpoint?

Reply:
I'm be confused in NewGRPCClient() there is no setting to track endpoint similar to [NewClient()](https://github.com/googleapis/google-cloud-go/blob/main/storage/storage.go#L185-L212). I can move in this block into NewGRPCClient() instead if that's what you're asking for. 

The endpoint is used to check if restricted.googleapis.com or private.googleapis.com is used for storage then monitoring API endpoint is changed to that.

Reply:
Discussed this offline; we should probably add a new func in google-api-go-client transport/http to just get the endpoint. @codyoss what do you think?

Reply:
Synced with @codyoss, he guided me through using manually initializing a ConnPool and passing it into the GAPIC client view grpc options. It does require duplicating default options from GAPIC client.

Reply:
Spoke too soon, I need the endpoint value before creating connPool; so will need to come up with a different solution. I'll file an issue for this; @tritone do you prefer one over the other in terms of less bad implementation? 

NewHTTPClient: https://github.com/googleapis/google-cloud-go/blob/1c537583857886881d3c3c87b8ad56d56cad22a1/storage/grpc_metrics.go#L233-L236
Double ConnPool: https://github.com/googleapis/google-cloud-go/pull/10639/commits/c0610c8cda5215cd93afda88b2ba1972b458122c

Reply:
Removed this entirely. no longer an issue.